src/models/schema.ts

```diff
@@ -91,6 +91,81 @@ export class Subject implements IResourceBase, ISearchable {
   searchableText(): string {
     return this.name;
   }
+
+  /**
+   * Merge the given schema array into ours, if any, ala mergesort:
+   * If we have NO schemas array currently, retain the provided array.
+   *
+   * If we have a schemas array, then retain any member that is present in the given array,
+   * as well as potentially add newly provided members.
+   */
+  mergeSchemas(newSchemas: Schema[]) {
+    // If we had no schemas previously, then graciously accept the new ones.
+    if (!this.schemas) {
+      this.schemas = newSchemas;
+      return;
+    }
+
+    // Iterate over both arrays in parallel. Both arrays are sorted by schema.version descending, so we can
+    // use a merge sort approach to combine them.
+    // 1. If a schema is present in both existing and new arrays, keep it in updated existing (merged).
+    // 2. If a schema is present in new array, but missing from existing, add it to merged.
+    // 3. If the schema is present in existing, but missing from new array, then skip it (do not add to merged).
```

Comment:
Seems like we could consolidate this whole section a bit -- what about keeping a `Map` of `version: schema` based on `this.schemas` and then just iterate through `newSchemas` to either return the existing schema version in the Map or the one in that given `newSchemas` iteration?

Reply:
Would also need a pass to ensure that any schema ONLY in original this.schemas gets _removed_ from end result if not also mentioned in the incoming array, then approaching this complexity.

I agree this reads a little complex, but am (hopefully correctly) confident that the test suite constrains its behavior properly.

Reply:
The operation here is akin to a `RIGHT OUTER JOIN`, choosing to include the elements in common (preferring the actual objects present on the left hand side -- `this.schemas`) and those present on the right hand side (passed-in `newSchemas`), but excluding those present only on the left hand side.